client/state/action-types.js

```diff
@@ -105,5 +109,22 @@ export const SUPPORT_USER_ACTIVATE = 'SUPPORT_USER_ACTIVATE';
 export const SUPPORT_USER_ERROR = 'SUPPORT_USER_ERROR';
 export const SUPPORT_USER_TOGGLE_DIALOG = 'SUPPORT_USER_TOGGLE_DIALOG';
 export const SUPPORT_USER_TOKEN_FETCH = 'SUPPORT_USER_TOKEN_FETCH';
+export const THEME_ACTIVATE = 'THEME_ACTIVATE';
+export const THEME_ACTIVATED = 'THEME_ACTIVATED';
+export const THEME_ACTIVATED_WITH_CUSTOMIZER = 'THEME_ACTIVATED_WITH_CUSTOMIZER';
+export const THEME_CLEAR_ACTIVATED = 'THEME_CLEAR_ACTIVATED';
+export const THEME_CUSTOMIZE = 'THEME_CUSTOMIZE';
+export const THEME_DETAILS_RECEIVE = 'THEME_DETAILS_RECEIVE';
+export const THEME_DETAILS = 'THEME_DETAILS';
```

Comment:
Yeah, good point. It navigates to a theme's details page -- so it's a side-effecting action, as you pointed out in https://github.com/Automattic/wp-calypso/issues/4057. It's very similar to `THEME_SIGNUP_WITH`, `THEME_PREVIEW`, or `THEME_CUSTOMIZE`. I think we made them actions mainy so we could tack tracking to them, but we probably don't do that anyway now.

I'm hesitant both to rename `THEME_DETAILS` to, say, `THEME_SHOW_DETAILS`, since it would break consistency with the others, and to remove those actions already in this PR, as that would add some more complexity and potential sources of errors.

Okay to leave? Probably going to try and rip them out in another PR.